utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/serialization/InstantSerializer.java

```diff
@@ -0,0 +1,30 @@
+package io.americanexpress.synapse.utilities.common.serialization;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+
+/**
+ * {@code InstantSerializer} serializes an {@link Instant} object into a string representing that single point on the timeline in ISO-8601 format in UTC.
+ */
+public class InstantSerializer extends JsonSerializer<Instant> {
+
+    private static final DateTimeFormatter ISO_INSTANT_OPTIONAL_SECOND_DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
+            .parseCaseInsensitive()
+            .append(DateTimeFormatter.ISO_LOCAL_DATE)
+            .appendLiteral('T')
+            .append(DateTimeFormatter.ISO_LOCAL_TIME)
+            .appendLiteral('Z')
+            .toFormatter()
+            .withZone(ZoneOffset.UTC);
```

Comment:
If we set the Zone, do we need to append the Literal `Z`?

Reply:
Yes we do otherwise, we end up getting the string back as `"2024-08-30T14:48:34.857239"` even though the java Instant object value was actually `2024-08-30T14:48:34.857239Z`. The numbers all look correct but we miss the Z if we remove the `appendLiteral`. If we remove the `withZone` then we get a parsing error entirely